### PR TITLE
fixes #12094 Added parent param to ensure pulp-nodes-parent is installed

### PR DESCRIPTION
**Pull request by johnpmitsch**

### CODE REVIEW

**Comment by johnpmitsch** [comment]
@ehelms I could be missing something, but I can't figure out where to put the test? Is manifest/install.pp a private class?

**Comment by ehelms** [comment]
Introduce a new spec - spec/classes/pulp_install_spec.rb and then look at (https://github.com/Katello/puppet-pulp/blob/master/spec/classes/pulp_config_spec.rb#L18) as an example. This is declaring the pulp main class, however, in your case you'd include the setting of `is_parent` to true.

File: spec/classes/pulp_install_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      :operatingsystemmajrelease  => '6.4',
+      :osfamily                   => 'RedHat',
+      :fqdn                       => 'pulp.compony.net',
+      :hostname                   => 'pulp',
+    }
+  end
+
+  describe "with is_parent" do
+    let :facts do
+      :default_facts
```
Comment (ehelms) [inline spec/classes/pulp_install_spec.rb]: No leading `:`

**Comment by johnpmitsch** [comment]
@ehelms updated, but still not sure why the test isn't passing :/

**Comment by ehelms** [comment]
You'll need to add this as a parameter on `init.pp` for users to be able to set when declaring the pulp class (as well as add a documentation entry)

**Comment by johnpmitsch** [comment]
@ehelms passing :)

File: manifests/init.pp [hunk 2]
```diff
@@ -205,7 +205,8 @@
 # $node_oauth_key::             The oauth key used to authenticate to the parent node
 #
 # $node_oauth_secret::          The oauth secret used to authenticate to the parent node
-#
+# 
+# $is_parent::                  Boolean whether the pulp install is a parent. Defaults to false
```
Comment (ehelms) [inline manifests/init.pp]: No need to include here what the default is -- params.pp handles that

File: manifests/init.pp [hunk 3]
```diff
@@ -276,6 +277,7 @@
   $node_oauth_effective_user = $pulp::params::node_oauth_effective_user,
   $node_oauth_key            = $pulp::params::node_oauth_key,
   $node_oauth_secret         = $pulp::params::node_oauth_secret,
+  $is_parent                 = $pulp::params::is_parent,
```
Comment (ehelms) [inline manifests/init.pp]: Can you add a validate_bool for this parameter?

**Comment by johnpmitsch** [comment]
@ehelms updated

**Comment by stbenjam** [comment]
Just my two cents, I would just to call it parent, we don't use the `is_` prefix for any other boolean that I know of.

**Comment by johnpmitsch** [comment]
@stbenjam updated

**Comment by stbenjam** [comment]
Thanks, much appreciated.  Ok from my side.

**Comment by ehelms** [comment]
ACK, thanks @johnpmitsch ! You can go ahead and open PRs to puppet-katello_devel and puppet-katello to set this parameter.

**Pull request merged**